### PR TITLE
React Native quickstart

### LINKED ISSUE #4702: Create a React Native example and Quickstart
# Chore

## Describe the chore

We already have an Expo guide here: 

And an example here: https://github.com/supabase/supabase/blob/master/examples/expo-todo-list

It would be great to have something similar for React Native. The code will be mostly the same, but it will be helpful for anyone who is new to RN and doesn't know how to translate the Expo examples to RN.

Todo:

- [x] example: https://github.com/supabase/supabase/pull/4733
- [ ] quickstart

### PR DESCRIPTION
**Pull request by Isaiah-Hamilton**
## What kind of change does this PR introduce?

Create React Native quickstart for the supabase docs.

## What is the current behavior?

Close #4702

## What is the new behavior?

React Native quickstart for the supabase docs.

## Additional context

Not sure if you wanted Expo React Native or plain React Native, just let me know. Also used a bit of @vasudeveloper001 [example code](https://github.com/supabase/supabase/tree/master/examples/ReactNativeTodoList) to speed up the process.

### CODE REVIEW

**Comment by Isaiah-Hamilton** [comment]
I'm not the best with React Native, so let me know if I need to change anything.

**Review by kiwicopple — APPROVED** [review]

**Comment by kiwicopple** [comment]
🔥  🔥

File: web/docs/guides/with-react-native.mdx [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+
+Let's start building the React Native app from scratch.
+
+### Initialize a React Native app
+
+We can use [`react-native`](https://reactnative.dev/blog/2016/11/08/introducing-button-yarn-and-a-public-roadmap#speed-up-react-native-init-using-yarn) to initialize 
+an app called `supabaseReactNative`:
+
+```bash
+expo init supabaseReactNative
```
Comment (kiwicopple) [inline web/docs/guides/with-react-native.mdx:151]: ah - I merged this before I realizing this is an expo setup. Can we please rename this one to "Expo Quickstart?

File: web/sidebars.js [hunk 2]
```diff
@@ -60,6 +60,7 @@ module.exports = {
         'guides/with-flutter',
         'guides/with-nextjs',
         'guides/with-react',
+        'guides/with-react-native',
```
Comment (kiwicopple) [inline web/sidebars.js:63]: let's change this one too!

**Pull request merged**